### ISSUE TITLE
Branch protection rules are not working anymore.

### ISSUE THREAD

**Issue by dgokcin** [repository-settings/app#625]
## Problem Description

I just created a new repository with the same config file that I use for all of my repositories and noticed that although the configs are the same, branch protection rules are not working. Any help is appreciated. To reproduce, you can try creating a new repo from the default settings in the README file. I am attaching my current config below

```yml
# These settings are synced to GitHub by https://probot.github.io/apps/settings/

repository:
  # See https://docs.github.com/en/rest/reference/repos#update-a-repository for all available settings.

  # The name of the repository. Changing this will rename the repository
  name: store-api

  # A short description of the repository that will show up on GitHub
  description: Store related apis

  # A URL with more information about the repository
  homepage: https://example.github.io/

  # A comma-separated list of topics to set on the repository
  topics: store, marketplace-integration, order

  # Either `true` to make the repository private, or `false` to make it public.
  private: true

  # Either `true` to enable issues for this repository, `false` to disable them.
  has_issues: false

  # Either `true` to enable projects for this repository, or `false` to disable them.
  # If projects are disabled for the organization, passing `true` will cause an API error.
  has_projects: false

  # Either `true` to enable the wiki for this repository, `false` to disable it.
  has_wiki: false

  # Either `true` to enable downloads for this repository, `false` to disable them.
  has_downloads: false

  # Updates the default branch for this repository.
  default_branch: develop

  # Either `true` to allow squash-merging pull requests, or `false` to prevent
  # squash-merging.
  allow_squash_merge: false

  # Either `true` to allow merging pull requests with a merge commit, or `false`
  # to prevent merging pull requests with merge commits.
  allow_merge_commit: true

  # Either `true` to allow rebase-merging pull requests, or `false` to prevent
  # rebase-merging.
  allow_rebase_merge: false

  # Either `true` to enable automatic deletion of branches on merge, or `false` to disable
  delete_branch_on_merge: false

  # Either `true` to enable automated security fixes, or `false` to disable
  # automated security fixes.
  enable_automated_security_fixes: false

  # Either `true` to enable vulnerability alerts, or `false` to disable
  # vulnerability alerts.
  enable_vulnerability_alerts: false

# Labels: define labels for Issues and Pull Requests
labels:
  - name: fix
    color: '#E1A296'
    description: A label used to mark fixes for automatic release notes generation.

  - name: feature
    # If including a `#`, make sure to wrap it with quotes!
    color: '#336699'
    description: A label used to mark features for automatic release notes generation.

# Collaborators: give specific users access to this repository.
# See https://docs.github.com/en/rest/reference/repos#add-a-repository-collaborator for available options
collaborators:
  # - username: bkeepers
  #   permission: push
  # - username: hubot
  #   permission: pull

  # Note: `permission` is only valid on organization-owned repositories.
  # The permission to grant the collaborator. Can be one of:
  # * `pull` - can pull, but not push to or administer this repository.
  # * `push` - can pull and push, but not administer this repository.
  # * `admin` - can pull, push and administer this repository.
  # * `maintain` - Recommended for project managers who need to manage the repository without access to sensitive or destructive actions.
  # * `triage` - Recommended for contributors who need to proactively manage issues and pull requests without write access.

# See https://docs.github.com/en/rest/reference/teams#add-or-update-team-repository-permissions for available options
teams:
  - name: CodeReview
    # The permission to grant the team. Can be one of:
    # * `pull` - can pull, but not push to or administer this repository.
    # * `push` - can pull and push, but not administer this repository.
    # * `admin` - can pull, push and administer this repository.
    # * `maintain` - Recommended for project managers who need to manage the repository without access to sensitive or destructive actions.
    # * `triage` - Recommended for contributors who need to proactively manage issues and pull requests without write access.
    permission: admin
  - name: DevTeam
    permission: push
  - name: PlatformTeam
    permission: admin

branches:
  - name: develop
    # https://docs.github.com/en/rest/reference/repos#update-branch-protection
    # Branch Protection settings. Set to null to disable
    protection:
      # Required. Require at least one approving review on a pull request, before merging. Set to null to disable.
      required_pull_request_reviews:
        # The number of approvals required. (1-6)
        required_approving_review_count: 1
        # Dismiss approved reviews automatically when a new commit is pushed.
        dismiss_stale_reviews: true
        # Blocks merge until code owners have reviewed.
        require_code_owner_reviews: true
        # Specify which users and teams can dismiss pull request reviews. Pass an empty dismissal_restrictions object to disable. User and team dismissal_restrictions are only available for organization-owned repositories. Omit this parameter for personal repositories.
        dismissal_restrictions:
          users: []
          teams: []
      # Required. Require status checks to pass before merging. Set to null to disable
      required_status_checks:
        # Required. Require branches to be up to date before merging.
        strict: true
        # Required. The list of status checks to require in order to merge into this branch
        contexts: ['Build & Test']
      # Required. Enforce all configured restrictions for administrators. Set to true to enforce required status checks for repository administrators. Set to null to disable.
      enforce_admins: null
      # Prevent merge commits from being pushed to matching branches
      required_linear_history: null
      # Required. Restrict who can push to this branch. Team and user restrictions are only available for organization-owned repositories. Set to null to disable.
      restrictions: null
  - name: main
    # https://docs.github.com/en/rest/reference/repos#update-branch-protection
    # Branch Protection settings. Set to null to disable
    protection:
      # Required. Require at least one approving review on a pull request, before merging. Set to null to disable.
      required_pull_request_reviews:
        # The number of approvals required. (1-6)
        required_approving_review_count: 1
        # Dismiss approved reviews automatically when a new commit is pushed.
        dismiss_stale_reviews: true
        # Blocks merge until code owners have reviewed.
        require_code_owner_reviews: true
        # Specify which users and teams can dismiss pull request reviews. Pass an empty dismissal_restrictions object to disable. User and team dismissal_restrictions are only available for organization-owned repositories. Omit this parameter for personal repositories.
        dismissal_restrictions:
          users: []
          teams: []
      # Required. Require status checks to pass before merging. Set to null to disable
      required_status_checks:
        # Required. Require branches to be up to date before merging.
        strict: true
        # Required. The list of status checks to require in order to merge into this branch
        contexts: ['Build & Test']
      # Required. Enforce all configured restrictions for administrators. Set to true to enforce required status checks for repository administrators. Set to null to disable.
      enforce_admins: null
      # Prevent merge commits from being pushed to matching branches
      required_linear_history: null
      # Required. Restrict who can push to this branch. Team and user restrictions are only available for organization-owned repositories. Set to null to disable.
      restrictions: null

```

**Comment by m-schrepel** [comment]
I am also having this issue with branch protection rules not being applied. I modified the settings.yml many times, checked the app installation many times and it is just not working.

**Comment by dgokcin** [comment]
any updates about this issue? it is a blocker on my way right now in using probot in all the repos in my organization. @travi sorry for mentioning you but I want to know the status of this issue since it has been almost a month. 

Thanks!

**Comment by travi** [comment]
sorry for the delay in investigating this, but i can confirm that branch protection itself is working correctly. however, as a result of https://github.com/repository-settings/app/pull/591, it is more likely applying branch protection rules will be missed since it is now only attempted after successful application of the other sections. 

this is especially unfortunate in the absence of https://github.com/repository-settings/app/issues/518, since it is quite difficult to understand a failure without running your own instance and having access to logs. if you are able to identify any problems with your configuration for the other sections, that is currently your best bet for unblocking branch protection settings from being applied. i hope to find some time to implement https://github.com/repository-settings/app/issues/518 soonish, which would provide far better insight into the details of config issues resulting in this blockage, but unfortunately i cannot make strong promises at this time.

**Comment by dgokcin** [comment]
@travi no problem. I do not understand what you meant by "applying branch protection rules will be missed since it is now only attempted after successful application of the other sections." you mean like I have something in my configuration which is causing a problem? in this case what do you think that might be?

thanks

**Comment by stevie-** [comment]
Is there a way to test a settings file? Like a github-action which parses the config in the same way as the probot-settings app.

**Comment by dgokcin** [comment]
@stevie- not that I know of. @travi my minimal configuration is prepared on top of the example in the repository README. I can confirm that only settings under branch protection rules are not working. All the other sections like repo description, labels, merge preferences etc. work.

How did you confirm that the branch protection rules were working on your test? Could you please share a config file?

Thanks

```
# These settings are synced to GitHub by https://probot.github.io/apps/settings/

repository:
  # See https://docs.github.com/en/rest/reference/repos#update-a-repository for all available settings.

  # The name of the repository. Changing this will rename the repository
  name: POCs

  # A short description of the repository that will show up on GitHub
  description: The source code for POCs.

  # A URL with more information about the repository
  homepage: https://example.github.io/

  # A comma-separated list of topics to set on the repository
  topics: poc, testing

  # Either `true` to make the repository private, or `false` to make it public.
  private: true

  # Either `true` to enable issues for this repository, `false` to disable them.
  has_issues: false

  # Either `true` to enable projects for this repository, or `false` to disable them.
  # If projects are disabled for the organization, passing `true` will cause an API error.
  has_projects: false

  # Either `true` to enable the wiki for this repository, `false` to disable it.
  has_wiki: false

  # Either `true` to enable downloads for this repository, `false` to disable them.
  has_downloads: false

  # Updates the default branch for this repository.
  default_branch: develop

  # Either `true` to allow squash-merging pull requests, or `false` to prevent
  # squash-merging.
  allow_squash_merge: false

  # Either `true` to allow merging pull requests with a merge commit, or `false`
  # to prevent merging pull requests with merge commits.
  allow_merge_commit: true

  # Either `true` to allow rebase-merging pull requests, or `false` to prevent
  # rebase-merging.
  allow_rebase_merge: false

  # Either `true` to enable automatic deletion of branches on merge, or `false` to disable
  delete_branch_on_merge: false

  # Either `true` to enable automated security fixes, or `false` to disable
  # automated security fixes.
  enable_automated_security_fixes: false

  # Either `true` to enable vulnerability alerts, or `false` to disable
  # vulnerability alerts.
  enable_vulnerability_alerts: false

branches:
  - name: develop
    # https://docs.github.com/en/rest/reference/repos#update-branch-protection
    # Branch Protection settings. Set to null to disable
    protection:
      # Required. Require at least one approving review on a pull request, before merging. Set to null to disable.
      required_pull_request_reviews:
        # The number of approvals required. (1-6)
        required_approving_review_count: 2
        # Dismiss approved reviews automatically when a new commit is pushed.
        dismiss_stale_reviews: true
        # Blocks merge until code owners have reviewed.
        require_code_owner_reviews: true
        # Specify which users and teams can dismiss pull request reviews. Pass an empty dismissal_restrictions object to disable. User and team dismissal_restrictions are only available for organization-owned repositories. Omit this parameter for personal repositories.
        dismissal_restrictions:
          users: []
          teams: []
      # Required. Require status checks to pass before merging. Set to null to disable
      required_status_checks:
        # Required. Require branches to be up to date before merging.
        strict: true
        # Required. The list of status checks to require in order to merge into this branch
        contexts: []
      # Required. Enforce all configured restrictions for administrators. Set to true to enforce required status checks for repository administrators. Set to null to disable.
      enforce_admins: null
      # Prevent merge commits from being pushed to matching branches
      required_linear_history: null
      # Required. Restrict who can push to this branch. Team and user restrictions are only available for organization-owned repositories. Set to null to disable.
      restrictions:
        apps: []
        users: []
        teams: []
```

**Comment by travi** [comment]
> you mean like I have something in my configuration which is causing a problem?

yes, the most likely situation would be that a call to update one of the other sections in your config is failing. this would result in the call to update branch protection not happening because it is only attempted after all of the other sections complete successfully. 

> in this case what do you think that might be?

this is why i mentioned it being unfortunate that #518 is not in place, since that would be able to provide feedback about the failure. solving that is high on my list, but i can't promise a timeline.

in my case, i had an issue with a color code for a label with leading zeros and worked around that in https://github.com/travi-test/.github/commit/17dde157c2d8d932cdf3c00009fa0bfb5fe339e2. i also implemented a more official fix for this in https://github.com/repository-settings/app/releases/tag/v2.0.3. in addition, i had some conflicts between my vulnerability alerts and security fixes settings, so i defined both as enabled in https://github.com/travi-test/.github/commit/9265ad34de79d71316eb72162009d7dd4fde6884 and https://github.com/travi-test/.github/commit/48fb2ca4ff8d5333ba22246319a37b66874cd602. 

> Could you please share a config file?

after making the above adjustments to my account-level config that is extended by my test project, i made the following change in order to re-trigger applying the project-level config and it applied the branch protection rules successfully: https://github.com/travi-test/settings-test/commit/16499fc1057af2f101d1ed867ea4c694f8ce2d7c

> my minimal configuration is prepared on top of the example in the repository README

i'll try to update my test config to be closer to yours when i get a chance and see if it results in any errors

**Comment by travi** [comment]
@dgokcin i just copy/pasted the config from your comment into https://github.com/travi-test/settings-test/commit/05de64cc77c934459b2ce48953013b471afb4374 and it applied successfully. 

i think i would recommend removing the other sections from your config file so that branches is the only section left. that should enable you to confirm if you have a problem with the branch protection section specifically or if the problem is coming from one of the other sections. assuming that confirms that branch protection is not the problem, you can then add the other sections back one by one, along with making a change to the branch protection at the same time. if the branch protection setting does not apply with adding one of the sections back to the file, that is the one causing the error. 

i know this is not ideal, but it is the best i can recommend until #518 is completed

**Comment by dgokcin** [comment]
after a painful hour of debugging, I found the problematic line. the ` enable_automated_security_fixes: false` line which is also on the README has caused  probot settings to fail processing at some point and eventually causing branch protection rules to fail. I am happy to say that removing that line fixed branch protection problems in all my repositories!

**Comment by dgokcin** [comment]
@travi thank you for your help, while going over your config, I noticed `_extends: .github`. how exactly does this work. could you please point me towards the correct documentation? I believe it allows inheriting config from a root repo?

**Comment by travi** [comment]
> I noticed `_extends: .github`. how exactly does this work. could you please point me towards the correct documentation?

this behavior is available from the probot framework used in this app and is mentioned in the readme for the project: https://github.com/repository-settings/app#inheritance

i'm going to hide this in this thread since it is a bit off topic, but feel free to open a new discussion if more clarification is needed on that topic

**Comment by dgokcin** [comment]
> > the ` enable_automated_security_fixes: false` line which is also on the README has caused probot settings to fail processing
> 
> would you be interested in sending a PR to update that property to `true` in the readme? probably a better recommendation anyway and is the setting that worked for me. before i set both security alerts and vulnerability fixes to true (which is really the behavior i wanted anyway), i was seeing "Vulnerability alerts must be enabled to configure automated security fixes.", so there are some requirements around those being used together.

it is already set to true in the readme. and my proobot config worked fine for a while. than all of a sudden, the same config stopped working. I believe setting both of them false maybe is breaking something.

**Comment by mdelapenya** [comment]
Unfortunately, I'm not able to make it work with this `settings.yml` file, and the branch protection only works if and only if it's alone in the file. I've tried with all permutations between major blocks (teams, collaborators, milestones, labels) without success. I wonder if it's possible to debug/log what section is the one failing.

The file:

```yaml
# These settings are synced to GitHub by https://probot.github.io/apps/settings/

repository:
  # See https://docs.github.com/en/rest/reference/repos#update-a-repository for all available settings.

  # The name of the repository. Changing this will rename the repository
  name: .github

  # A short description of the repository that will show up on GitHub
  description: Base repository

  # A URL with more information about the repository
  homepage: https://www.example.com

  # A comma-separated list of topics to set on the repository
  topics: settings

  # Either `true` to make the repository private, or `false` to make it public.
  private: false

  # Either `true` to enable issues for this repository, `false` to disable them.
  has_issues: true

  # Either `true` to enable projects for this repository, or `false` to disable them.
  # If projects are disabled for the organization, passing `true` will cause an API error.
  has_projects: true

  # Either `true` to enable the wiki for this repository, `false` to disable it.
  has_wiki: true

  # Either `true` to enable downloads for this repository, `false` to disable them.
  has_downloads: false

  # Updates the default branch for this repository.
  default_branch: main

  # Either `true` to allow squash-merging pull requests, or `false` to prevent
  # squash-merging.
  allow_squash_merge: true

  # Either `true` to allow merging pull requests with a merge commit, or `false`
  # to prevent merging pull requests with merge commits.
  allow_merge_commit: false

  # Either `true` to allow rebase-merging pull requests, or `false` to prevent
  # rebase-merging.
  allow_rebase_merge: false

  # Either `true` to enable automatic deletion of branches on merge, or `false` to disable
  delete_branch_on_merge: true

  # Either `true` to enable automated security fixes, or `false` to disable
  # automated security fixes.
  enable_automated_security_fixes: false

  # Either `true` to enable vulnerability alerts, or `false` to disable
  # vulnerability alerts.
  enable_vulnerability_alerts: false

# Labels: define labels for Issues and Pull Requests
# If including a `#`, make sure to wrap it with quotes!
labels:
  - name: type/bug
    color: '#CC0000'
    description: An issue with the system.

  - name: type/feature
    color: '#336699'
    description: New functionality.

  - name: low-hanging-fruit
    color: '#FFC300'
    description: Issues that are good for newcomers.

  - name: hacktoberfest-accepted
    color: '#ff0000'
    description: Pull Requests accepted for Hacktoberfest.

  - name: invalid
    color: '#C3C3C3'
    description: Not accepted or invalid for any reason.

# Collaborators: give specific users access to this repository.
# See https://docs.github.com/en/rest/reference/repos#add-a-repository-collaborator for available options
collaborators:
  - username: foo
    permission: admin
  - username: bar
    permission: triage
  - username: baaz
    permission: admin

  # Note: `permission` is only valid on organization-owned repositories.
  # The permission to grant the collaborator. Can be one of:
  # * `pull` - can pull, but not push to or administer this repository.
  # * `push` - can pull and push, but not administer this repository.
  # * `admin` - can pull, push and administer this repository.
  # * `maintain` - Recommended for project managers who need to manage the repository without access to sensitive or destructive actions.
  # * `triage` - Recommended for contributors who need to proactively manage issues and pull requests without write access.

# See https://docs.github.com/en/rest/reference/teams#add-or-update-team-repository-permissions for available options
teams:
  # Please make sure the team already exist in the organization, as the repository-settings application is not creating them.
  # See https://github.com/repository-settings/app/discussions/639 for more information about teams and settings
  - name: core-team
    # The permission to grant the team. Can be one of:
    # * `pull` - can pull, but not push to or administer this repository.
    # * `push` - can pull and push, but not administer this repository.
    # * `admin` - can pull, push and administer this repository.
    # * `maintain` - Recommended for project managers who need to manage the repository without access to sensitive or destructive actions.
    # * `triage` - Recommended for contributors who need to proactively manage issues and pull requests without write access.
    permission: admin
  - name: a-team
    permission: pull
  - name: b-team
    permission: push

branches:
  - name: main
    # https://docs.github.com/en/rest/reference/repos#update-branch-protection
    # Branch Protection settings. Set to null to disable
    protection:
      # Required. Require at least one approving review on a pull request, before merging. Set to null to disable.
      required_pull_request_reviews:
        # The number of approvals required. (1-6)
        required_approving_review_count: 1
        # Dismiss approved reviews automatically when a new commit is pushed.
        dismiss_stale_reviews: true
        # Blocks merge until code owners have reviewed.
        require_code_owner_reviews: true
        # Specify which users and teams can dismiss pull request reviews. Pass an empty dismissal_restrictions object to disable. User and team dismissal_restrictions are only available for organization-owned repositories. Omit this parameter for personal repositories.
        dismissal_restrictions:
          users: []
          teams: []
      # Required. Require status checks to pass before merging. Set to null to disable
      required_status_checks:
        # Required. Require branches to be up to date before merging.
        strict: true
        # Required. The list of status checks to require in order to merge into this branch
        contexts: []
      # Required. Enforce all configured restrictions for administrators. Set to true to enforce required status checks for repository administrators. Set to null to disable.
      enforce_admins: true
      # Prevent merge commits from being pushed to matching branches
      required_linear_history: true
      # Required. Restrict who can push to this branch. Team and user restrictions are only available for organization-owned repositories. Set to null to disable.
      restrictions:
        apps: []
        users: []
        teams: []
```

**Comment by dgokcin** [comment]
@mdelapenya 

what happens if you delete just the following lines?

'''
enable_automated_security_fixes: false

  # Either `true` to enable vulnerability alerts, or `false` to disable
  # vulnerability alerts.
  enable_vulnerability_alerts: false
'''

**Comment by mdelapenya** [comment]
> @mdelapenya
> 
> what happens if you delete just the following lines?
> 
> enable_automated_security_fixes: false enable_vulnerability_alerts: false

Unfortunately nothing. See: https://github.com/tc-experiments/.github/blob/main/.github/settings.yml The branch protection is not created

**Comment by arrase** [comment]
same problem here, branch protection is not created

**Comment by benjyiw** [comment]
FWIW, I was able to fix the branch protection problem in my case by adjusting the default labels. I was using the label configuration exactly as is from the README from this project. This sounds pretty silly, but I tried removing the caterpillar emoji, and it seemed to have fixed the problem in my repos.

Orig from README:
```
labels:
  - name: bug
    color: CC0000
    description: An issue with the system 🐛.
```

And now:
```
labels:
  - name: bug
    color: CC0000
    description: An issue with the system.
```

Additionally, in my case I already had all of the security settings enabled, which others mentioned was the culprit for their problems. So that didn't seem to be causing an issue for me. Hope this information is helpful to others.

**Comment by rogerluan** [comment]
In my case it was definitely [this](https://github.com/rogerluan/.github/commit/b6e4f1ba587e3bc176e77b28abc8066c8d8a8536) setting that was malfunctioning in my repo, but perhaps **also** [this](https://github.com/rogerluan/.github/commit/33c700895bc3e4135d11a4e7fa94929d1971fbe3) one, the reason being (for both of them) that I am running this all on personal repositories. Hope this helps someone 💪  My tip here is to read the comments to determine whether you should be enabling a setting or not, and then also read the documentation specified in each section (that leads to github official doc websites), those helped me a lot.

**Comment by rogerluan** [comment]
In another repo of mine, [this](https://github.com/rogerluan/FeatureConfig/commit/671d713c53755f9b6605d389482767dd492a60a1) was the root cause of the failure. Apparently, topics cannot have spaces in them 😬

**Comment by mdelapenya** [comment]
Mmm, interestingly, I had an issue with the teams: there were teams that did not existed, and until creating them, the configuration failed.

History:
- https://github.com/tc-experiments/.github/commit/52ba68fb7d505360b0137bfd89bf68a5f6bf6d49 worked ✅ 
- Added the GH teams that were missing
- https://github.com/tc-experiments/.github/commit/d4f6742638491c0ef7eb67034aabcf9131deaf3a failed 🔴 

Permalink to the settings that worked: https://github.com/tc-experiments/.github/blob/52ba68fb7d505360b0137bfd89bf68a5f6bf6d49/.github/settings.yml

**Comment by stevie-** [comment]
We see issues when we set `teams` parameter. If I omit it, branch protection rules work. If I add teams, teams will be set, but branch protection is skipped.

**Comment by kmgrime** [comment]
Got this to work after some debugging and a lot of testing we had to change or remove the following stuff.  

1. color: CC0000 **<---- change this to '#CC0000'**
2. description: An issue with the system 🐛. **<---- remove the emoji**
3.  name: core **<---- make sure it is the exact same name as it is in you github org and don't include org name.**
 branches:
4.   name: master **<---- change to main as main is GitHubs default and master is the old default**
    
5. **#Dont change any properties under protection except values (true / false etc), it needs the properties to parse correctly it seems. Use the one in the readme for template but fix the values pointed out**
    
protection:
      # Required. Require at least one approving review on a pull request, before merging. Set to null to disable.
      required_pull_request_reviews:
        # The number of approvals required. (1-6)
        required_approving_review_count: 1
        # Dismiss approved reviews automatically when a new commit is pushed.
        dismiss_stale_reviews: true
        # Blocks merge until code owners have reviewed.
        require_code_owner_reviews: true
        # Specify which users and teams can dismiss pull request reviews. Pass an empty dismissal_restrictions object to disable. User and team dismissal_restrictions are only available for organization-owned repositories. Omit this parameter for personal repositories.
        dismissal_restrictions:
          users: []
          teams: []
      # Required. Require status checks to pass before merging. Set to null to disable
      required_status_checks:
        # Required. Require branches to be up to date before merging.
        strict: true
        # Required. The list of status checks to require in order to merge into this branch
        contexts: []
      # Required. Enforce all configured restrictions for administrators. Set to true to enforce required status checks for repository administrators. Set to null to disable.
      enforce_admins: true
      # Prevent merge commits from being pushed to matching branches
      required_linear_history: true
      # Required. Restrict who can push to this branch. Team and user restrictions are only available for organization-owned repositories. Set to null to disable.
      restrictions:
        apps: []
        users: []
        teams: []

**Comment by wknapik** [comment]
Ran into this as well. Had a team with the org prefix in settings.yml and branch protection rules after that. Neither the team, nor branch protection settings were applied, until I commented out the `teams` section.

> name: core <---- make sure it is the exact same name as it is in you github org and don't include org name.

@travi how do we reference a team from another org? We have multiple orgs and would like to reference teams across them.

**Comment by mrlesmithjr** [comment]
I have also spent the past day trying to figure this out but have been unsuccessful. Is it safe to assume that setting these rules is completely broken for now?

**Comment by kmgrime** [comment]
> Ran into this as well. Had a team with the org prefix in settings.yml and branch protection rules after that. Neither the team, nor branch protection settings were applied, until I commented out the `teams` section.
> 
> > name: core <---- make sure it is the exact same name as it is in you github org and don't include org name.
> 
> @travi how do we reference a team from another org? We have multiple orgs and would like to reference teams across them.

Have you tried ex:

teams:
  - name: org-name/team-read

Its very important that the .YML file is correctly structured with the right amount of indents / indentations is correct. This goes for all .yml files in my experience.

**Comment by kmgrime** [comment]
> I have also spent the past day trying to figure this out but have been unsuccessful. Is it safe to assume that setting these rules is completely broken for now?

Could you paste your full settings.yml file here and maybe I can help as I was stuck several times myself and got to solve it. 
Make sure to replace company namings and naming conventions with some random stuff. 👍

**Comment by mdelapenya** [comment]
A working `settings.yml` that works for us: https://github.com/testcontainers/testcontainers-go/blob/main/.github/settings.yml

Hopefully it helps 🤞

**Comment by mrlesmithjr** [comment]
@kmgrime - Here is one that I've tried

https://github.com/mrlesmithjr/ansible-nginx/blob/master/.github/settings.yml

**Comment by kmgrime** [comment]
> Here is one that I've tried
> 
> https://github.com/mrlesmithjr/ansible-nginx/blob/master/.github/settings.yml

@mrlesmithjr - If this is the file you are running you don't have all the settings required in the file so it will not be parsed correctly. If you have settings that you don't use in the file, leave them empty instead of removing them. 
There is multiple good settings files here that is complete that you can use as template from the answers in this thread, hopefully one of them could be a good starting point. 
Bottom line, all settings need to be in the file for it to work, even if you don't need them - leave them blank. 

Good luck 👍

**Comment by dworac** [comment]
Hey everyone! I've been having this issue for the past couple of hours and it's driving me crazy so I decided to run my own instance to be able to look at logs:

Aside from the already mentioned problems other users suggested like @kmgrime or @benjyiw my problem was fixed after removing the `dismissal_restrictions` property. 

As you can see in the logs: 

> Validation Failed: \"Only organization repositories can have users and team restrictions\"

There might be other properties affecting your `settings.yml` file related to this so be sure to take a good look.

I will finally suggest to run your own instance of the bot as it helps a lot finding the particual issue in your file. There is a good guide in the [Docs deploying guide](https://github.com/repository-settings/app/blob/master/docs/deploy.md)

![image](https://user-images.githubusercontent.com/117940053/211124545-221e0532-ad76-4ea9-a7ba-7fd8ec2ab76f.png)

**Comment by Shahard2** [comment]
Is there a real fix for it?

I've tried all the things that people here mentioned.. but nothing works, Branch protection is not created.

**Comment by Shahard2** [comment]
@travi

**Comment by mike-carey** [comment]
We had similar issues where the settings were working and then one day they stopped working for certain repos.  After attempting many of the fixes suggested above, we were able to resolve our issue by omitting the `enable_automated_security_fixes` property.  Whether we enable or disable, just having this property present seems to cause an issue.  Looking at our other repos that the settings were still being applied for, the `enable_automated_security_fixes` was not in the settings file and thus the protections were still being applied.  All other properties seem to be working fine for us including the `enable_vulnerability_alerts` which seems strange to me since they use the same logic.

**Comment by Shahard2** [comment]
> We had similar issues where the settings were working and then one day they stopped working for certain repos. After attempting many of the fixes suggested above, we were able to resolve our issue by omitting the `enable_automated_security_fixes` property. Whether we enable or disable, just having this property present seems to cause an issue. Looking at our other repos that the settings were still being applied for, the `enable_automated_security_fixes` was not in the settings file and thus the protections were still being applied. All other properties seem to be working fine for us including the `enable_vulnerability_alerts` which seems strange to me since they use similar logic in the code.


Thank you mate,
Tried to remove enable_automated_security_fixes

Still not working for me :/

**Comment by mike-carey** [comment]
> We had similar issues where the settings were working and then one day they stopped working for certain repos. After attempting many of the fixes suggested above, we were able to resolve our issue by omitting the `enable_automated_security_fixes` property. Whether we enable or disable, just having this property present seems to cause an issue. Looking at our other repos that the settings were still being applied for, the `enable_automated_security_fixes` was not in the settings file and thus the protections were still being applied. All other properties seem to be working fine for us including the `enable_vulnerability_alerts` which seems strange to me since they use similar logic in the code.

Looks like I was wrong about what was happening here.  Setting `enable_automated_security_fixes` to true while `enable_vulnerability_alerts` is false seems to be the issue we had.  I cannot even get a description update when I have that combination of settings.

```yaml
---
repository:
  # ...
  enable_automated_security_fixes: true
  enable_vulnerability_alerts: false
```

^^ This does not work

**Comment by mike-carey** [comment]
Sorry one more update.  It does look like when I set both `enable_automated_security_fixes` and `enable_vulnerability_alerts` to false, the branch protections do not get updated, but the description does get updated.  Commenting out the `enable_automated_security_fixes` is what gets our branch protections to be updated.

```yaml
---
repository:
  # ...
  enable_automated_security_fixes: false
  enable_vulnerability_alerts: false
```

^^ This also does not work.

If you are going to disable security features, I would leave the `enable_automated_security_fixes` out completely.

---

## To summarize:

```yaml
---
repository:
  # ...
  description: 'This has been changed'   # this change will not take effect
  enable_automated_security_fixes: true
  enable_vulnerability_alerts: false
branches:
  - name: main
    protection:
      # ...
      required_pull_request_reviews:
        required_approving_review_count: 2  # this change will not take effect
```

Neither the description nor the branch protection gets updated.

```yaml
---
repository:
  # ...
  description: 'This has been changed'  # this change will take effect
  enable_automated_security_fixes: true
  enable_vulnerability_alerts: false
branches:
  - name: main
    protection:
      # ...
      required_pull_request_reviews:
        required_approving_review_count: 2  # this change will not take effect
```

The description is updated, but the branch protection does not get updated.

**Comment by Shahard2** [comment]
> guys, is settings app dead?

looks like.......

**Comment by kmgrime** [comment]
> > [Starting September 20th, 2023, users can configure their workflows using rulesets in order to run and pass in selected repositories before merging their code. On October 18th, users will no longer be able to access Actions Required Workflows and must use rulesets in its place.](https://github.blog/changelog/2023-08-02-github-actions-required-workflows-will-move-to-repository-rules/)
> 
> probably will affect the settings app

@electriquo This ^

**Comment by hwhsu1231** [comment]
Any progress of this issue? I encounter this issue as well, and couldn't solve it. The branch protection rule cannot be created by `.github/settings.yml` after trying all the possible solutions mentioned above.

**Comment by travi** [comment]
trying to clean up some duplicates, so closing this in favor of https://github.com/repository-settings/app/issues/150

short answer is that this does still work when the config is all formatted correctly and no other failures have occurred ahead of the attempt to apply the branch protection settings. https://github.com/repository-settings/app/issues/518 needs to be implemented to at least provide feedback when failures occur to give an opportunity to resolve any config problems
